pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -341,6 +342,20 @@ func (c *Controller) onServiceEvent(curr interface{}, event model.Event) error {
 		c.Unlock()
 	}
 
+	// Update endpoint shards when service added, in case endpoint added earlier than service
+	if event == model.EventAdd {
+		// Build IstioEndpoints
+		endpoints := c.endpoints.buildIstioEndpointsWithService(svc.Name, svc.Namespace, svcConv.Hostname)
```

Comment:
Is it possible to check if endpoints for this service already exist in ep cache - possibly expose a method in endpoint controller and only trigger this when they are already there?

Reply:
It might not be that impactful because we are only doing that for new service - but just want to check?

Reply:
In theory, we can do that. Given to the complexity, I gave up.

Reply:
Ok. that is fine. Can you change log.Debugf("Get endpoint %s/%s failed %v", name, namespace, err) and remove "failed" similar to how you did for ep slice?

Reply:
ok